### ISSUE TITLE
Authorize codes via OAuth with expiry

### ISSUE THREAD

**Issue by robnagler** [radiasoft/sirepo#4106]
Allow code roles to be authorized via OAuth. Sirepo will ask the OAuth server to validate user credentials. The OAuth server will return an access token with an expiry. 

The OAuth handoff will need to describe where it is going so that needs to be configurable on a per code basis.

Roles need an expiry and a description of an authentication method. I don't now that we can reuse sirepo.auth for this, but this is so specific that I think we can just refactor sirepo.auth.github module to support this workflow.

**Comment by e-carlin** [comment]
Want to confirm that we are going to do the full authorization code grant type (read [here](https://www.digitalocean.com/community/tutorials/an-introduction-to-oauth-2) for exactly what I mean) where we will redirect to a flash page, the user will authenticate, and then will be redirected back to sirepo.com?

**Comment by robnagler** [comment]
That is correct. They need to enter their credentials with the 3rd party provider (e.g. GitHub) and then come back to us.
